src/Utility/dontYield.lua

```diff
@@ -0,0 +1,52 @@
+--!nonstrict
+
+--[[
+    Utility function to throw an error when a function yields
+
+    You can either provide a specific error message using the second
+    parameter (errorMessage: string?), or you can let it default to the
+    default error message (DEFAULT_ERROR_MESSAGE). Error message here
+    means additional information that the user might find helpful to
+    determine where the yielding is occurring. (i.e. "Cannot yield inside
+    of the ComputedPairs processor function.",)
+]]
+
+local Package = script.Parent.Parent
+local logError = require(Package.Logging.logError)
+
+local DEFAULT_ERROR_MESSAGE_ID: string = "cannotYield"
+
+--[[
+    Handles the return values resulting from calling `coroutine.resume(func, ...)`
+    Makes it so we don't have to use table.unpack to get the Variant returned from
+    calling `coroutine.resume(func, ...)`
+]]
+local function handleResult(taskCoroutine, messageID: string?, success, ...)
+    if success then
+        if coroutine.status(taskCoroutine) == "dead" then
+            -- we didn't yield
+            return ...
+            
+        else
+            -- we yielded
+            logError(messageID or DEFAULT_ERROR_MESSAGE_ID, ({...})[1])
+        end
+    else
+        -- this is a user error that has nothing to do with the thread yielding,
+        -- so error like the user would normally expect it to
+        error(debug.traceback(taskCoroutine, ...), 2)
```

Comment:
error level should probably be 3

Reply:
1 - the line calling error
2 - the Fusion function calling dontYield
3 - Could be the user's code, but might also be another Fusion function that called 2? I'm not sure 3 will always work.

Reply:
I know what error levels are. You probably don't want to error inside the dontYield function. you want to error at the function calling dontYield. 2 isn't super appropriate because this is an internal function; errors with stack levels (intended to be from the user) should usually error across the module boundary (aka into whatever called a modulescript's function), which would be 3 in this case.

Reply:
But 3 might also be another Fusion function that called level 2. I'm saying 3 might not always be far enough to trace back to the user.

Reply:
That's fine, it still communicates more information. To be perfectly clear the "user" of dontYield is whatever called the function. Pointing to the call site of dontYield is the most useful thing to do here. Not some line inside dontYield.

Reply:
I agree, error level 3 communicates more accurate information. If this gets approved for v.0.3 I'll make the change, for now I'm leaving this conversation unresolved.